src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -262,6 +331,69 @@ export default {
 
             this.time = this.timeFormat.format(datum);
             this.imageUrl = this.imageFormat.format(datum);
+        },
+        getFormatter(key) {
+            let valueFormatter = this.openmct.telemetry.getValueFormatter({
+                format: key
+            });
+
+            return valueFormatter.formatter;
+        },
+        numericImageAge() {
+            let currentTime = this.currentTimeValue();
+            let selectedImage = this.getSelectedImage();
+            let formatter = this.getFormatter(this.timeKey);
+
+            if (selectedImage === undefined || selectedImage[this.timeKey] === undefined) {
+                return;
+            }
+
+            let parsedSelectedTime = formatter.parse(selectedImage[this.timeKey]);
+
+            return currentTime - parsedSelectedTime;
+        },
+        formattedAge() {
+            let age = this.numericImageAge();
+
+            if (!age) {
+                return;
+            }
+
+            let result = this.durationFormatter.format(age);
+
+            if (age > EIGHT_HOURS) {
+                let negativeAge = (age / ONE_HOUR) * -1;
+                result = moment.duration(negativeAge, 'hours').humanize(true);
+
+                if (age > TWENTYFOUR_HOURS) {
+                    negativeAge = (age / TWENTYFOUR_HOURS) * -1;
+                    result = moment.duration(negativeAge, 'days').humanize(true);
+                }
+            }
+
+            return result;
+        },
+        trackAge() {
+            let age;
+            this.clearAgeTracking();
+            this.ageTracker = window.setInterval(() => {
+                age = this.formattedAge();
+                if (age) {
+                    this.age = age;
+                } else {
+                    this.age = 'N/A';
+                }
+            }, AGE_TRACK_INTERVAL_MS);
+        },
+        clearAgeTracking() {
+            window.clearInterval(this.ageTracker);
+        },
+        resetAgeCSS() {
+            this.refreshCSS = true;
+            window.setTimeout(() => {
```

Comment:
This is an unfortunately consequence of CSS animations. The problem here is that we need to clear the CSS class, wait a bit, and then set again in order to trigger the animation. Using a flag to clear it is a reasonable approach. Since it depends upon Vue re-evaluating the flag in order to clear the CSS class a better way to approach this might be to use `Vue.nextTick()`. That way it is guaranteed that Vue has reset the flag.

eg.
``` JavaScript
this.refreshCSS = true;
Vue.nextTick(() => {
    this.refreshCSS = false;
});
```

Reply:
Nice, good call.